prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -114,6 +118,10 @@ class FHIRTranslator(
         logger.trace("Preparing to send original message")
         val originalReport = reportService.getRootReport(message.reportId)
         val bodyBytes = BlobAccess.downloadBlobAsByteArray(originalReport.bodyUrl)
+        val localDigest = BlobUtils.digestToString(sha256Digest(bodyBytes))
+        check(message.digest == localDigest) {
```

Comment:
This seems wrong to me. The `digest` value in the message I believe would be the digest of the blob associated with the passed in report (so the report that is downloaded in `sendTranslated`). Here, you are downloading the ROOT report, which would have a different digest.

Reply:
So, if I'm understanding correctly, we should be comparing message.digest with blobInfo.digest (after the file is uploaded)?

Reply:
`message.digest` is already used in `sendTranslated` and that's the only place it should be used. Here, you are downloading the root report, so you need to to compare the downloaded root report to the root report's digest (which is stored in the DB report_file table)

Reply:
Latest changes should address this.